modules/installing-gcp-user-defined-labels-and-tags.adoc

```diff
@@ -0,0 +1,126 @@
+
+:_content-type: CONCEPT
+[id="installing-gcp-user-defined-labels-and-tags_{context}"]
+= Configuring the user-defined labels and tags for GCP
+
+
+GCP provides tags that help to identify and organize the resources created for a specific {product-title} cluster, making them easier to manage. 
+
+You can define labels & tags on the GCP resources only during cluster creation that is available from {product-title} version {product-version}. User-defined labels and tags are not supported for {product-title} clusters upgraded to {product-title} version {product-version}.
+
+User-defined labels and {product-title} specific labels are applied only to resources created by {product-title} installer and its core operators such as `machine-api-provider-gcp`, `cluster-image-registry-operator`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` and not on the resources created by any other operators or the kubernetes in-tree components that creates resources for example the ingress load balancers.
+
+User-defined tags are attached to resources created by {product-title} installer and its core operators such as `cluster-image-registry-operator` and not on the resources created by any other operators or the kubernetes in-tree components. In {product-title} version {product-version}, resources created by `machine-api-provider-gcp`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` are not tagged.
+
+[NOTE]
+====
+Labels & tags once defined during the install-config creation phase cannot be modified or updated with new labels & tags after cluster creation.
+====
+
+Support for user-defined labels & tags for GCP resources is available in {product-title} version {product-version} as Technology Preview (TP).
+
+Service Account used for cluster creation must have the `TagUser` role assigned to it, required for tags defined at both organization and project level.
+
+You can use `.platform.gcp.userLabels` and `platform.gcp.userTags` fields to define the list of desired labels & tags in `install-config.yaml`. `.platform.gcp.userLabels` and `.platform.gcp.userTags` fields are a yaml array of objects type.
+
+.Sample `install-config.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+featureSet: TechPreviewNoUpgrade
+platform:
+ gcp:
+   userLabels:
+   - key: <label_key>
+     value: <label_value>
+   userTags:
+   - parentID: <OrganizationID/ProjectID>
+     key: <tag_key_short_name>
+     value: <tag_value_short_name>
+----
+
+The user-defined labels have the following requirements:
+
+* A label key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A label key and value must contain only lowercase letters, numeric characters, underscores, and dashes.
+* A label key must start with a lowercase letter.
+* Each resource can have a maximum of 32 labels, and 32 labels are reserved for internal use by {product-title}.
+* You can configure a maximum of 64 labels per resource.
```

Comment:
Sorry, I'm confused a bit with the above 2 statements (line 49 and line 50), e.g. how many user-defined labels can be specified at most, in `platform.gcp.userLabels` of `install-config.yaml`? Thanks!

Reply:
The draft content was: 

* Each resource can have multiple labels, up to a maximum of 32.

**  GCP supports a maximum of 64 labels per resource. Restricting the number of user defined labels to 32 and reserving 32 for Openshift's internal use.

I will change the content to -
* You can configure a maximum of 64 labels per resource.
** Each resource can have a maximum of 32 labels, and 32 labels are reserved for internal use by {product-title}.

Reply:
Seems like the meaning that needs to be delivered through these steps is:
* Each resource can have multiple labels, up to a maximum of 32.
** GCP supports a maximum of 64 labels per resource. Restricting the number of user defined labels to 32 and reserving 32 for Openshift's internal use.

So, I will have to revert the changes back.